sdk/client/src/blocking.rs

```diff
@@ -233,6 +238,54 @@ impl BlockingClient {
         self.send(MethodRequest::get_events_with_proofs(key, start_seq, limit))
     }
 
+    /// Return the events of type `T` that have been emitted to `event_key` since `start_seq`, with a max of `limit`
+    /// results
+    /// Returns an empty vector if there are no such events
+    pub fn get_deserialized_events<T: MoveResource + DeserializeOwned>(
+        &self,
+        event_key: &EventKey,
+        start_seq: u64,
+        limit: u64,
+    ) -> Result<Response<Vec<Event<T>>>> {
+        let (events, state) = self
+            .get_events_with_proofs(&hex::encode(event_key.as_bytes()), start_seq, limit)?
```

Comment:
oops I must have forgotten to change the API of `get_events_with_proofs` so that it takes a typed EventKey instead of a string, which will eliminate the need to do this.